src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0012Tests.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Xunit;
+using Verifier = Azure.ClientSdk.Analyzers.Tests.AzureAnalyzerVerifier<Azure.ClientSdk.Analyzers.TypeNameAnalyzer>;
+
+namespace Azure.ClientSdk.Analyzers.Tests
+{
+    public class AZC0012Tests
+    {
+        [Fact]
+        public async Task AZC0001ProducedForSingleWordTypeNames()
+        {
+            const string code = @"
+namespace Azure.Data
+{
+    public class [|Program|] { }
```

Comment:
Should we check for something like `IProgram`?

Reply:
Yes, good point.